### ISSUE TITLE
CPU可以训练这个模型吗？

### ISSUE THREAD

**Issue by csn223355** [hustvl/YOLOP#167]
老师，作者你们好，我想用笔记本电脑的cpu试着运行一下训练的代码。结果报错，错误如下：
(YOLOP) csn@dominant-yoga-4:~/Project/VScode/python/YOLOP$ python tools/train.py
=> creating runs/BddDataset/_2022-11-22-16-42
Namespace(conf_thres=0.001, dataDir='', iou_thres=0.6, local_rank=-1, logDir='runs/', modelDir='', prevModelDir='', sync_bn=False)
AUTO_RESUME: False
CUDNN:
  BENCHMARK: True
  DETERMINISTIC: False
  ENABLED: True
DATASET:
  COLOR_RGB: False
  DATAROOT: /home/csn/Dataset/BDD/BDD100K/images
  DATASET: BddDataset
  DATA_FORMAT: jpg
  FLIP: True
  HSV_H: 0.015
  HSV_S: 0.7
  HSV_V: 0.4
  LABELROOT: /home/csn/Dataset/BDD/BDD100K/det_annotations
  LANEROOT: /home/csn/Dataset/BDD/BDD100K/ll_seg_annotations
  MASKROOT: /home/csn/Dataset/BDD/BDD100K/da_seg_annotations
  ORG_IMG_SIZE: [720, 1280]
  ROT_FACTOR: 10
  SCALE_FACTOR: 0.25
  SELECT_DATA: False
  SHEAR: 0.0
  TEST_SET: val
  TRAIN_SET: train
  TRANSLATE: 0.1
DEBUG: False
GPUS: (0, 1)
LOG_DIR: runs/
LOSS:
  BOX_GAIN: 0.05
  CLS_GAIN: 0.5
  CLS_POS_WEIGHT: 1.0
  DA_SEG_GAIN: 0.2
  FL_GAMMA: 0.0
  LL_IOU_GAIN: 0.2
  LL_SEG_GAIN: 0.2
  LOSS_NAME: 
  MULTI_HEAD_LAMBDA: None
  OBJ_GAIN: 1.0
  OBJ_POS_WEIGHT: 1.0
  SEG_POS_WEIGHT: 1.0
MODEL:
  EXTRA:
    
  HEADS_NAME: ['']
  IMAGE_SIZE: [640, 640]
  NAME: 
  PRETRAINED: 
  PRETRAINED_DET: 
  STRU_WITHSHARE: False
NEED_AUTOANCHOR: False
PIN_MEMORY: False
PRINT_FREQ: 20
TEST:
  BATCH_SIZE_PER_GPU: 24
  MODEL_FILE: 
  NMS_CONF_THRESHOLD: 0.001
  NMS_IOU_THRESHOLD: 0.6
  PLOTS: True
  SAVE_JSON: False
  SAVE_TXT: False
TRAIN:
  ANCHOR_THRESHOLD: 4.0
  BATCH_SIZE_PER_GPU: 24
  BEGIN_EPOCH: 0
  DET_ONLY: False
  DRIVABLE_ONLY: True
  ENC_DET_ONLY: False
  ENC_SEG_ONLY: False
  END_EPOCH: 240
  GAMMA1: 0.99
  GAMMA2: 0.0
  IOU_THRESHOLD: 0.2
  LANE_ONLY: False
  LR0: 0.001
  LRF: 0.2
  MOMENTUM: 0.937
  NESTEROV: True
  OPTIMIZER: adam
  PLOT: True
  SEG_ONLY: False
  SHUFFLE: True
  VAL_FREQ: 1
  WARMUP_BIASE_LR: 0.1
  WARMUP_EPOCHS: 3.0
  WARMUP_MOMENTUM: 0.8
  WD: 0.0005
WORKERS: 8
num_seg_class: 2
begin to bulid up model...
/home/csn/anaconda3/envs/YOLOP/lib/python3.7/site-packages/torch/cuda/__init__.py:52: UserWarning: CUDA initialization: Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx (Triggered internally at  /opt/conda/conda-bld/pytorch_1603729138878/work/c10/cuda/CUDAFunctions.cpp:100.)
  return torch._C._cuda_getDeviceCount() > 0
Using torch 1.7.0 CPU

load model to device
freeze encoder and Det head and Ll_Seg heads...
freezing model.0.conv.conv.weight
freezing model.0.conv.bn.weight
freezing model.0.conv.bn.bias
freezing model.1.conv.weight
freezing model.1.bn.weight
freezing model.1.bn.bias
freezing model.2.cv1.conv.weight
freezing model.2.cv1.bn.weight
freezing model.2.cv1.bn.bias
freezing model.2.cv2.weight
freezing model.2.cv3.weight
freezing model.2.cv4.conv.weight
freezing model.2.cv4.bn.weight
freezing model.2.cv4.bn.bias
freezing model.2.bn.weight
freezing model.2.bn.bias
freezing model.2.m.0.cv1.conv.weight
freezing model.2.m.0.cv1.bn.weight
freezing model.2.m.0.cv1.bn.bias
freezing model.2.m.0.cv2.conv.weight
freezing model.2.m.0.cv2.bn.weight
freezing model.2.m.0.cv2.bn.bias
freezing model.3.conv.weight
freezing model.3.bn.weight
freezing model.3.bn.bias
freezing model.4.cv1.conv.weight
freezing model.4.cv1.bn.weight
freezing model.4.cv1.bn.bias
freezing model.4.cv2.weight
freezing model.4.cv3.weight
freezing model.4.cv4.conv.weight
freezing model.4.cv4.bn.weight
freezing model.4.cv4.bn.bias
freezing model.4.bn.weight
freezing model.4.bn.bias
freezing model.4.m.0.cv1.conv.weight
freezing model.4.m.0.cv1.bn.weight
freezing model.4.m.0.cv1.bn.bias
freezing model.4.m.0.cv2.conv.weight
freezing model.4.m.0.cv2.bn.weight
freezing model.4.m.0.cv2.bn.bias
freezing model.4.m.1.cv1.conv.weight
freezing model.4.m.1.cv1.bn.weight
freezing model.4.m.1.cv1.bn.bias
freezing model.4.m.1.cv2.conv.weight
freezing model.4.m.1.cv2.bn.weight
freezing model.4.m.1.cv2.bn.bias
freezing model.4.m.2.cv1.conv.weight
freezing model.4.m.2.cv1.bn.weight
freezing model.4.m.2.cv1.bn.bias
freezing model.4.m.2.cv2.conv.weight
freezing model.4.m.2.cv2.bn.weight
freezing model.4.m.2.cv2.bn.bias
freezing model.5.conv.weight
freezing model.5.bn.weight
freezing model.5.bn.bias
freezing model.6.cv1.conv.weight
freezing model.6.cv1.bn.weight
freezing model.6.cv1.bn.bias
freezing model.6.cv2.weight
freezing model.6.cv3.weight
freezing model.6.cv4.conv.weight
freezing model.6.cv4.bn.weight
freezing model.6.cv4.bn.bias
freezing model.6.bn.weight
freezing model.6.bn.bias
freezing model.6.m.0.cv1.conv.weight
freezing model.6.m.0.cv1.bn.weight
freezing model.6.m.0.cv1.bn.bias
freezing model.6.m.0.cv2.conv.weight
freezing model.6.m.0.cv2.bn.weight
freezing model.6.m.0.cv2.bn.bias
freezing model.6.m.1.cv1.conv.weight
freezing model.6.m.1.cv1.bn.weight
freezing model.6.m.1.cv1.bn.bias
freezing model.6.m.1.cv2.conv.weight
freezing model.6.m.1.cv2.bn.weight
freezing model.6.m.1.cv2.bn.bias
freezing model.6.m.2.cv1.conv.weight
freezing model.6.m.2.cv1.bn.weight
freezing model.6.m.2.cv1.bn.bias
freezing model.6.m.2.cv2.conv.weight
freezing model.6.m.2.cv2.bn.weight
freezing model.6.m.2.cv2.bn.bias
freezing model.7.conv.weight
freezing model.7.bn.weight
freezing model.7.bn.bias
freezing model.8.cv1.conv.weight
freezing model.8.cv1.bn.weight
freezing model.8.cv1.bn.bias
freezing model.8.cv2.conv.weight
freezing model.8.cv2.bn.weight
freezing model.8.cv2.bn.bias
freezing model.9.cv1.conv.weight
freezing model.9.cv1.bn.weight
freezing model.9.cv1.bn.bias
freezing model.9.cv2.weight
freezing model.9.cv3.weight
freezing model.9.cv4.conv.weight
freezing model.9.cv4.bn.weight
freezing model.9.cv4.bn.bias
freezing model.9.bn.weight
freezing model.9.bn.bias
freezing model.9.m.0.cv1.conv.weight
freezing model.9.m.0.cv1.bn.weight
freezing model.9.m.0.cv1.bn.bias
freezing model.9.m.0.cv2.conv.weight
freezing model.9.m.0.cv2.bn.weight
freezing model.9.m.0.cv2.bn.bias
freezing model.10.conv.weight
freezing model.10.bn.weight
freezing model.10.bn.bias
freezing model.13.cv1.conv.weight
freezing model.13.cv1.bn.weight
freezing model.13.cv1.bn.bias
freezing model.13.cv2.weight
freezing model.13.cv3.weight
freezing model.13.cv4.conv.weight
freezing model.13.cv4.bn.weight
freezing model.13.cv4.bn.bias
freezing model.13.bn.weight
freezing model.13.bn.bias
freezing model.13.m.0.cv1.conv.weight
freezing model.13.m.0.cv1.bn.weight
freezing model.13.m.0.cv1.bn.bias
freezing model.13.m.0.cv2.conv.weight
freezing model.13.m.0.cv2.bn.weight
freezing model.13.m.0.cv2.bn.bias
freezing model.14.conv.weight
freezing model.14.bn.weight
freezing model.14.bn.bias
freezing model.17.cv1.conv.weight
freezing model.17.cv1.bn.weight
freezing model.17.cv1.bn.bias
freezing model.17.cv2.weight
freezing model.17.cv3.weight
freezing model.17.cv4.conv.weight
freezing model.17.cv4.bn.weight
freezing model.17.cv4.bn.bias
freezing model.17.bn.weight
freezing model.17.bn.bias
freezing model.17.m.0.cv1.conv.weight
freezing model.17.m.0.cv1.bn.weight
freezing model.17.m.0.cv1.bn.bias
freezing model.17.m.0.cv2.conv.weight
freezing model.17.m.0.cv2.bn.weight
freezing model.17.m.0.cv2.bn.bias
freezing model.18.conv.weight
freezing model.18.bn.weight
freezing model.18.bn.bias
freezing model.20.cv1.conv.weight
freezing model.20.cv1.bn.weight
freezing model.20.cv1.bn.bias
freezing model.20.cv2.weight
freezing model.20.cv3.weight
freezing model.20.cv4.conv.weight
freezing model.20.cv4.bn.weight
freezing model.20.cv4.bn.bias
freezing model.20.bn.weight
freezing model.20.bn.bias
freezing model.20.m.0.cv1.conv.weight
freezing model.20.m.0.cv1.bn.weight
freezing model.20.m.0.cv1.bn.bias
freezing model.20.m.0.cv2.conv.weight
freezing model.20.m.0.cv2.bn.weight
freezing model.20.m.0.cv2.bn.bias
freezing model.21.conv.weight
freezing model.21.bn.weight
freezing model.21.bn.bias
freezing model.23.cv1.conv.weight
freezing model.23.cv1.bn.weight
freezing model.23.cv1.bn.bias
freezing model.23.cv2.weight
freezing model.23.cv3.weight
freezing model.23.cv4.conv.weight
freezing model.23.cv4.bn.weight
freezing model.23.cv4.bn.bias
freezing model.23.bn.weight
freezing model.23.bn.bias
freezing model.23.m.0.cv1.conv.weight
freezing model.23.m.0.cv1.bn.weight
freezing model.23.m.0.cv1.bn.bias
freezing model.23.m.0.cv2.conv.weight
freezing model.23.m.0.cv2.bn.weight
freezing model.23.m.0.cv2.bn.bias
freezing model.24.m.0.weight
freezing model.24.m.0.bias
freezing model.24.m.1.weight
freezing model.24.m.1.bias
freezing model.24.m.2.weight
freezing model.24.m.2.bias
freezing model.34.conv.weight
freezing model.34.bn.weight
freezing model.34.bn.bias
freezing model.36.cv1.conv.weight
freezing model.36.cv1.bn.weight
freezing model.36.cv1.bn.bias
freezing model.36.cv2.weight
freezing model.36.cv3.weight
freezing model.36.cv4.conv.weight
freezing model.36.cv4.bn.weight
freezing model.36.cv4.bn.bias
freezing model.36.bn.weight
freezing model.36.bn.bias
freezing model.36.m.0.cv1.conv.weight
freezing model.36.m.0.cv1.bn.weight
freezing model.36.m.0.cv1.bn.bias
freezing model.36.m.0.cv2.conv.weight
freezing model.36.m.0.cv2.bn.weight
freezing model.36.m.0.cv2.bn.bias
freezing model.37.conv.weight
freezing model.37.bn.weight
freezing model.37.bn.bias
freezing model.39.conv.weight
freezing model.39.bn.weight
freezing model.39.bn.bias
freezing model.40.cv1.conv.weight
freezing model.40.cv1.bn.weight
freezing model.40.cv1.bn.bias
freezing model.40.cv2.weight
freezing model.40.cv3.weight
freezing model.40.cv4.conv.weight
freezing model.40.cv4.bn.weight
freezing model.40.cv4.bn.bias
freezing model.40.bn.weight
freezing model.40.bn.bias
freezing model.40.m.0.cv1.conv.weight
freezing model.40.m.0.cv1.bn.weight
freezing model.40.m.0.cv1.bn.bias
freezing model.40.m.0.cv2.conv.weight
freezing model.40.m.0.cv2.bn.weight
freezing model.40.m.0.cv2.bn.bias
freezing model.42.conv.weight
freezing model.42.bn.weight
freezing model.42.bn.bias
begin to load data
building database...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 70000/70000 [00:30<00:00, 2263.05it/s]
database build finish
building database...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10000/10000 [00:04<00:00, 2325.73it/s]
database build finish
load data finished
anchors loaded successfully
tensor([[[0.3750, 1.1250],
         [0.6250, 1.3750],
         [0.5000, 2.5000]],

        [[0.4375, 1.1250],
         [0.3750, 2.4375],
         [0.7500, 1.9375]],

        [[0.5938, 1.5625],
         [1.1875, 2.5312],
         [2.1250, 4.9062]]])
=> start training...
Killed
请问如何解决这个问题。

**Comment by csn223355** [comment]
已经解决该问题，原因是bach_size太大了，调小一些就好。
